pkg/ingester/ingester.go

```diff
@@ -99,7 +101,10 @@ type Config struct {
 
 	WAL WALConfig `yaml:"wal,omitempty" doc:"description=The ingester WAL (Write Ahead Log) records incoming logs and stores them on the local file systems in order to guarantee persistence of acknowledged data in the event of a process crash."`
 
-	ChunkFilterer chunk.RequestChunkFilterer `yaml:"-"`
+	ChunkFilterer          chunk.RequestChunkFilterer     `yaml:"-"`
```

Comment:
It's quite weird to see these objects in a `Config`, but I see they are not the only ones like that :confused: Is that the only way these non-config objects can be injected? :confused:

Reply:
It's not the only way, just the way they happen to be injected in the ingester. I've followed the pattern here rather than bifurcating how it's wired together.